Support separate WebUI user / view modes

spacebares opened the pull request:
looks like this:
https://github.com/blakeblackshear/frigate/assets/57186372/46937caa-a7a9-4661-a6d7-4dd427295787

default behavior can be set like
```
ui:
  show_advanced_options: True
```

spacebares committed:
toggle viewing of advanced options

- fix AppBar render order, its tooltips are visible now

spacebares commented:
lint complains about `web/src/context/index.jsx
  19:6  warning  React Hook useEffect has a missing dependency: 'showAdvOptions'. Either include it or remove the dependency array` but if i add that in then when you refresh page it forgets cookie. 

maybe this whole thing is coded wrong, i haven't a clue, react is a wonder to behold...

NickM-27 commented:
I think this is better presented as `Admin Mode` and the configuration option can be `default_mode` with options `admin` or `user`

spacebares commented:
that would imply more than simple on/off. Could be user/advanced/admin/debug etc.
To support multiple view options have to rewrite this, i have some ideas.. maybe a new component that hides its children and it does all the work in there. So if you want something controlled by the view mode you just throw it in the component as a child.. i think that might work. Will try some things.

spacebares committed:
support multiple viewmodes for UI options and menus

spacebares commented:
some wip on many options:
https://github.com/blakeblackshear/frigate/assets/57186372/202a2a2a-a62a-4f6f-b4e0-a0ebe4883405

spacebares commented on web/src/context/index.jsx:
```diff
@@ -1,6 +1,30 @@
+    }
+
+    load();
+  }, [setViewMode, config]);
```
yea but when i do poot it in there, everything stops working 🤷‍♂️🤷‍♂️🤷‍♂️🤷‍♂️

NickM-27 reviewed (commented):
Very cool, I think that's what Blake had in mind and it's great how easily it can be expanded to other things in the future

spacebares commented on web/src/components/AppBar.jsx:
```diff
@@ -38,7 +38,7 @@ export default function AppBar({ title: Title, overflowRef, onOverflowClick }) {
   return (
     <div
       id="appbar"
-      className={`w-full border-b border-gray-200  dark:border-gray-700 flex items-center align-middle p-2  fixed left-0 right-0 z-10 bg-white dark:bg-gray-900 transform transition-all duration-200 ${
```
to fix tooltips in the appbar 
![image](https://github.com/blakeblackshear/frigate/assets/57186372/56759696-3890-4ab4-ac88-6d008ae8b9b1)

Had to also move `AppBar` farther down to get it to render on top of everything [web/src/app.tsx line 28](https://github.com/blakeblackshear/frigate/blob/fff14f7dc61382804f7a27e333d2e8fbbef6d681/web/src/app.tsx#L28C27-L28C27).

spacebares committed:
fix ui viewmode button look

spacebares commented:
menu looks like this atm:
![image](https://github.com/blakeblackshear/frigate/assets/57186372/c0d8ecf6-cf65-4967-bb55-b837d3a90741)

but i was thinking about:
![1ZDSi6ZJzG](https://github.com/blakeblackshear/frigate/assets/57186372/8326d0ef-8144-41d1-b36b-ac0cada83357)

not sure which one to pick, 🤔

NickM-27 commented:
I prefer the former, makes it much more obvious which is the currently selected mode and takes up less space

spacebares committed:
revert appbar draworder

the event thumbnail renders over AppBar in events page... tried a bunch of things, counlt work around it...

with this revert, the "More Options" tooltip for AppBar menu is broken again

spacebares committed:
fix rare config load bug

- view mode will default to admin if config does fail to load for w/e reason
- rename viewMode to currentViewMode

spacebares committed:
fix web tests

spacebares committed:
Merge branch 'dev' of https://github.com/blakeblackshear/frigate into advancedoptionsui

spacebares committed:
only load config value if no cookie

spacebares committed:
web tests for viewmode

spacebares committed:
ViewMode renamed to UserView in code

spacebares committed:
docs for viewmode in the config

spacebares committed:
Merge branch 'dev' into advancedoptionsui

johnmarksilly commented:
Dropdown inside a menu is not ideal though. A separate icon button for "user pref" might be better. Profile icon in header expands to the view modes.

NickM-27 commented:
Thank you for your pull request, we are currently rewriting the UI for the next version of frigate and this PR can't be merged. This functionality is being considered for the new UI.
